### PR TITLE
Custom filename for POST file upload

### PR DESCRIPTION
**Pull request by SamuelePilleri**
Added support for custom filenames in POST request when uploading files through HTTP.

Example:
```
.setOpt(Curl.option.HTTPPOST, [
	{ name : "foo", contents : "bar" },
	{
		name     : "Filedata",
		file     : "/path/to/audio/with/custom/name.custom_ext",
		filename : "audio.mp3", // as it appears in post
		type     : "audio/mpeg"
	}
])
```

### CODE REVIEW

**Comment by JCMais** [comment]
Awesome!

Can you add one test for given upload with filename?

It would be inside this file https://github.com/JCMais/node-libcurl/blob/develop/test/curl/httppost.js, and mostly identical to the ``should upload file correctly`` test, but with the postData different. As postData is currently a global variable ([:35](https://github.com/JCMais/node-libcurl/blob/develop/test/curl/httppost.js#L35)), make it local to each test. It's basically the only thing going to change.

Also, squash your changes into a single commit. Thanks! :+1:

File: test/curl/httppost.js [hunk 1]
```diff
@@ -121,3 +122,42 @@ it( 'should upload file correctly', function ( done ) {
...
+        if ( status !== 200 ) {
+            throw Error( 'Invalid status code: ' + status );
+        }
+
+        data = JSON.parse( data );
+
+        data.size.should.be.equal( size );
+        //data.name.should.be.equal( imageFilename.split( path.sep ).pop() );
+        data.type.should.be.equal( postData[0].type );
+        data.filename.should.be.equal( _name );
```
Comment (JCMais) [inline test/curl/httppost.js]: Remove this line.

``data`` here is the one returned on line [:71](https://github.com/SamuelePilleri/node-libcurl/blob/develop/test/curl/httppost.js#L71). So you just need to check for the name field to be equals the _name variable (line 148 above that you commented).

**Comment by SamuelePilleri** [comment]
Done. Let's see the tests :)

**Comment by JCMais** [comment]
Thanks!

**Comment by SamuelePilleri** [comment]
That's cool! When is it going to land on npm?

**Comment by JCMais** [comment]
I can try to release a new version Friday, if I can't, this weekend

**Comment by SamuelePilleri** [comment]
Awesome, thank you!                                                                                                                                                                                                                 Da: Jonathan Cardoso MachadoInviato: mercoledì 13 aprile 2016 16:59A: JCMais/node-libcurlRispondi a: JCMais/node-libcurlCc: Samuele PilleriOggetto: Re: [JCMais/node-libcurl] Custom filename for POST file upload (#55)I can try to release a new version Friday, if I can't, this weekend

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub

**Pull request merged**